ci/kokoro/docker/build-in-docker-bazel.sh

```diff
@@ -85,12 +85,6 @@ echo "bazel test" "${bazel_args[@]}"
 "${BAZEL_BIN}" test \
   "${bazel_args[@]}" "--test_tag_filters=-integration-test" ...
 
-echo "================================================================"
-io::log "Compiling all the code, including integration tests"
-# Then build everything else (integration tests, examples, etc). So we can run
-# them next.
-"${BAZEL_BIN}" build "${bazel_args[@]}" ...
```

Comment:
AFAIK all the other `"${BAZEL_BIN}"` commands in this script filter out *something*.  This is (or was), the only command that guaranteed everything built.  Did I miss something?  If I did, just merge and please explain it to me later, no worries.

Reply:
Just a few lines above this we have

https://github.com/googleapis/google-cloud-cpp/blob/e102e5ee35304723716fad99a3b2b1e0c738e20c/ci/kokoro/docker/build-in-docker-bazel.sh#L85-L86

which will do the following:

* It will [build everything](https://docs.bazel.build/versions/master/user-manual.html#test) matching `...` (i.e., everything)
* It will run all the tests, except for those with the `integration-test` tag

So in the best case, the `bazel build` command that this PR removes would do nothing (because the previous command already built everything). But in the worst case, this bazel build will recompile everything again if the flags changed between the "test" and the "build". The flags don't actually change currently, so today, this bazel build is simply doing nothing. However, _iff_ that `bazel test` command were changed to `bazel coverage` (as I'm experimenting with in https://github.com/googleapis/google-cloud-cpp/pull/5065), then the compile flags _will_ be different, and this `bazel build` command will needlessly recompile all targets.

Regardless of whether we move forward w/ the other PR to use `bazel coverage`, it seems like this `bazel build` step is unnecessary.

Reply:
Doh, I missed your earlier answer last night, sorry for the noise.